pallets/settlement/src/benchmarking.rs

```diff
@@ -0,0 +1,568 @@
+
+        let l in 0 .. T::MaxLegsInInstruction::get() as u32;
+        let mut legs: Vec<Leg<T::Balance>> = Vec::with_capacity(l as usize);
+        let mut portfolios: Vec<PortfolioId> = Vec::with_capacity(l as usize);
+        // create venue
+        let Account {account_id, origin, did} = make_account::<T>("creator", SEED);
+        let venue_id = create_venue_::<T>(did, vec![]);
+        for n in 1 .. l {
+            setup_leg_and_portfolio::<T>(None, Some(did), n, &mut legs, &mut portfolios)?;
+        }
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id: u64 = 1;
+        // Affirm an instruction
+        let portfolios_set = portfolios.clone().into_iter().collect::<BTreeSet<_>>();
+        Module::<T>::unsafe_affirm_instruction(did, instruction_id, portfolios_set)?;
+
+    }: _(origin, instruction_id, portfolios)
+    verify {
+        for (idx, leg) in legs.iter().enumerate() {
+            ensure!(matches!(Module::<T>::instruction_leg_status(instruction_id, u64::try_from(idx).unwrap_or_default()), LegStatus::PendingTokenLock), "Fail: withdraw affirmation dispatch");
+        }
+    }
+
+
+    withdraw_affirmation_with_receipt {
+        // Below setup is for the receipt based affirmation
+
+        let l in 0 .. T::MaxLegsInInstruction::get() as u32;
+        let mut legs: Vec<Leg<T::Balance>> = Vec::with_capacity(l as usize);
+        let mut portfolios: Vec<PortfolioId> = Vec::with_capacity(l as usize);
+        // create venue
+        let Account {account_id, origin, did} = make_account::<T>("creator", SEED);
+        let venue_id = create_venue_::<T>(did, vec![account_id.clone()]);
+        for n in 1 .. l {
+            setup_leg_and_portfolio::<T>(None, Some(did), n, &mut legs, &mut portfolios)?;
+        }
+        // Add instruction
+        Module::<T>::base_add_instruction(did, venue_id, SettlementType::SettleOnAffirmation, None, legs.clone())?;
+        let instruction_id: u64 = 1;
+        // Affirm an instruction
+        portfolios.clone().into_iter().for_each(|p| {
+            set_user_affirmations(instruction_id, p, AffirmationStatus::Affirmed);
+        });
+        for (idx, _) in legs.clone().iter().enumerate() {
+            let leg_id = u64::try_from(idx).unwrap_or_default();
+            // use leg_id for the receipt_uid as well.
+            set_instruction_let_status_to_skipped::<T>(instruction_id, leg_id, account_id.clone(), leg_id);
+        }
+    }: withdraw_affirmation(origin, instruction_id, portfolios)
+    verify {
+        for (idx, leg) in legs.iter().enumerate() {
+            ensure!(matches!(Module::<T>::instruction_leg_status(instruction_id, u64::try_from(idx).unwrap_or_default()), LegStatus::PendingTokenLock), "Fail: withdraw affirmation dispatch");
+        }
+    }
+
+    withdraw_affirmation_with_both_receipt_and_onchain_affirmation {
```

Comment:
This is not needed. We'll be using weight for whatever is more expensive (receipts vs no-receipts).